docs/guides/src/main/server/configuration_overview.adoc

```diff
@@ -0,0 +1,173 @@
+a lot of processing that would usually happen when starting and running the server is no longer necessary and the server can start and run faster.
+
+Some optimizations performed by the `build` command are:
+
+* Closed-world assumption about installed providers, hence no need to re-create the registry every time the server starts
+* Configuration files are pre-parsed to reduce IO when starting the server
+* Database specific resources are configured and prepared to run against a specific database vendor
+* By persisting build options into the server image, the server does not perform any additional step to interpret configuration options and (re)configure itself
+
+Once you run the `build` command, you won't need to set the same build options again (they are now persisted) so that you can just `start` the server:
+
+.First stage, run the `build` command to set any build option
+<@kc.build parameters="--db=postgres"/>
+
+.Second stage, run the `start` command to set any configuration option when starting the server
+<@kc.start parameters="--db-url-host=keycloak-postgres --db-username=keycloak --db-password=change_me --hostname mykeycloak.acme.com"/>
+
+== Configuring the server for an optimal startup time
+
+In addition to the optimizations performed when you run the `build` command, you might want to avoid using CLI options when running the
+`start` command in favor of either using environment variables or configuration options from the `conf/keycloak.conf` file. By favoring the last two configuration approaches, the server is going to skip some steps at startup and start faster.
+
+.Set any build option
+<@kc.build parameters="--db=postgres"/>
+
+.Set any configuration option to `conf/keycloak.conf`
+```
+db-url-host=keycloak-postgres
+db-username=keycloak
+db-password=change_me
+hostname mykeycloak.acme.com
+```
+
+.Start the server
+<@kc.start/>
+
+== Configuring the server from a properties file
+
+By default, the server is going to always fetch any configuration option you set in the `conf/keycloak.conf` file. When you are using a fresh distribution,
+this file holds only the recommended settings for running in production which are initially commented out.
+
+You can also specify a different configuration file by using the `[-cf|--config-file] option as follows:
+
+<@kc.all rootParameters="-cf myconfig.conf"/>
+
+[NOTE]
+Changes to any configuration option defined in the `keycloak.conf` file that is targeted for the `build` command is ignored
+if the value differs from the value used to previously run the `build` command. In this case, make sure you run the `build` command again so that
+these configurations options take effect.
+
+== Using placeholders
+
+You should be able to use placeholders to dynamically change the value of configuration options when starting the server.
+
+For that, you should use the following format when defining the value of an option:
```

Comment:
Try to follow "this" or "that" with a noun ....   For that purpose, 
Also, generally convert "You should use" to "Use..."  It means the same thing and is more concise.